### ISSUE TITLE
Expose Type Mapping for Object Columns in System Tables (e.g., information_schema.columns)

### ISSUE THREAD

**Issue by proddata** [crate/crate#16662]
### Problem Statement

Currently, there is no direct way to retrieve the mapping type (ignored, dynamic, or strict) of object columns from system tables like information_schema.columns. The only way to obtain this information is through the SHOW CREATE TABLE command, which is not convenient for automated systems or more advanced querying needs.

### Possible Solutions

Expose the mapping type (ignored, dynamic, strict) for object columns in information_schema.columns or another relevant system table. This would provide easier access to this information for users who need to query or programmatically manage their table schemas.

**Comment by mfussenegger** [comment]
>  which is not convenient for automated systems or more advanced querying needs.

Do you have a concrete example for where you need to query the information?

**Comment by proddata** [comment]
> Do you have a concrete example for where you need to query the information?

Yes, this information is necessary to determine if an object needs to be transformed before insertion due to dynamic typing, or if it can be passed directly into an ignored object column. Examples include heterogeneous arrays or properties with varying sub-types.

Another use case being discussed is giving users the ability to display and potentially modify the table schema in a user interface. Currently, this is not possible without parsing the CREATE TABLE statement.
